client/src/js/services/grid/TransactionService.js

```diff
@@ -303,24 +303,81 @@ function TransactionService($timeout, util, uiGridConstants, bhConstants, Notify
    * This method is used to check if a transaction is balanced by their record
    */
   function validateTransaction(entity) {
-    var transaction = entity.data.data;
-    var debit = 0,
-      credit = 0;
-    
-    transaction.forEach(function (row) {
-      debit += Number(row.debit_equiv);
-      credit += Number(row.credit_equiv);
+    var transaction = [];
+
+    // include transaction data
+    entity.data.data.forEach(function (row) {
+      transaction.push(row);
     });
 
-    var ERR_UNBALANCED_TXN = 'POSTING_JOURNAL.ERRORS.UNBALANCED_TRANSACTIONS';
+    // include new rows
+    entity.newRows.data.forEach(function (row) {
+      transaction.push(row);
+    });
 
-    // later in validateTransaction()
+    var numberOfLine = transaction.length;
     var error;
 
-    if (debit !== credit) {
-      error = ERR_UNBALANCED_TXN;
+    // If the transaction are single line transaction 
+    if(numberOfLine === 1){
+      error = 'POSTING_JOURNAL.ERRORS.SINGLE_LINE_TRANSACTION';
+    } else {
```

Comment:
Instead of doing this giant `if () else {}`, you can just return here - if there is only one line, it is clearly a single line transaction and we don't need to do anything else.  It will save you one indentation.

Like this:
```js
// as suggested earlier...
var ERR_SINGLE_LINE_TRANSACTION = 'POSTING_JOURNAL.ERRORS.SINGLE_LINE_TRANSACTION';

// and then here....

if (numberOfLines === 1) {
  return ERR_SINGLE_LINE_TRANSACTION;
}

var debit = 0;
var credit = 0;
/// ... etc...
```